roles/rhel_subscribe/tasks/main.yml

```diff
@@ -0,0 +1,49 @@
+---
+# TODO: Enhance redhat_subscription module
+#       to make it able to attach to a pool
+#       to make it able to enable repositories
+
+- set_fact:
+    rhel_subscription_user: "{{ lookup('env', 'rhel_subscription_user') | default('None', True) }}"
+    rhel_subscription_pass: "{{ lookup('env', 'rhel_subscription_pass') | default('None', True) }}"
```

Comment:
Something like the following will prevent having to do as many when checks below.  It also fits in with our new style guide (https://github.com/openshift/openshift-ansible/blob/master/docs/best_practices_guide.adoc#defensive-programming).  This also introduces a variable that can be set in the user's inventory or custom playbooks as well to set the rhsm user/password.
```
set_fact:
  rhel_subscription_user: "{{ lookup('env', 'rhel_subscription_user') | default(rhsub_user, True) }}"
  rhel_subscription_pass: "{{ lookup('env', 'rhel_subscription_pass') | default(rhsub_pass, True) }}"
fail:
  msg: "This role is only supported for Red Hat hosts"
  when: ansible_distribution != 'RedHat'
fail:
  msg: Eiither rsub_user or the rhel_subscription_user env variable are required for this role.
  when: not (rhel_subscription_user is defined and rhel_subscription_user | bool)
fail:
  msg: Eiither rsub_pass or the rhel_subscription_pass env variable are required for this role.
  when: not (rhel_subscription_pass is defined and rhel_subscription_pass | bool)
```